conn.go

```diff
@@ -59,23 +69,109 @@ func (cc *Conn) Flush() error {
 
 	defer conn.Close()
 
-	if _, err := conn.SendMessages(batch(cc.messages)); err != nil {
+	cc.endBatch(cc.messages)
+
+	_, err = conn.SendMessages(cc.messages[:cc.it+1])
+
+	if err != nil {
 		return fmt.Errorf("SendMessages: %w", err)
 	}
 
-	if _, err := conn.Receive(); err != nil {
-		return fmt.Errorf("Receive: %w", err)
+	// Retrieving of seq number associated to entities
+	entitiesBySeq := make(map[uint32]Entity)
+	for i, e := range cc.entities {
+		entitiesBySeq[cc.messages[i].Header.Sequence] = e
+	}
+
+	// Trigger entities callback
+	msg, err := cc.checkReceive(conn)
+	for msg {
+		rmsg, err := conn.Receive()
+
+		if err != nil {
+			return fmt.Errorf("Receive: %w", err)
+		}
+
+		for _, msg := range rmsg {
+			if e, ok := entitiesBySeq[msg.Header.Sequence]; ok {
+				e.HandleResponse(msg)
+			}
+		}
+		msg, err = cc.checkReceive(conn)
+	}
+
+	return err
+}
+
+// PutMessage store netlink message to sent after
+func (cc *Conn) PutMessage(msg netlink.Message) int32 {
```

Comment:
Can you un-export putMessage? It doesn’t need to be public API, right?

Reply:
Right